ReactAndroid/src/main/java/com/facebook/react/bridge/ReadableNativeArray.java

```diff
@@ -46,4 +49,38 @@ protected ReadableNativeArray(HybridData hybridData) {
   public native ReadableNativeMap getMap(int index);
   @Override
   public native ReadableType getType(int index);
+
+  public List<Object> toJavaArray() {
+    return ReadableNativeArray.toJavaArray(this);
+  }
+
+  public static List<Object> toJavaArray(ReadableNativeArray readableArray) {
+    List<Object> unwrappedList = new ArrayList<>(readableArray.size());
+    for (int i = 0; i < readableArray.size(); i++) {
+      ReadableType type = readableArray.getType(i);
+      switch (type) {
+        case Null:
+          unwrappedList.add(i, null);
+          break;
+        case Boolean:
+          unwrappedList.add(i, readableArray.getBoolean(i));
+          break;
+        case Number:
+          unwrappedList.add(i, readableArray.getDouble(i));
+          break;
+        case String:
+          unwrappedList.add(i, readableArray.getString(i));
+          break;
+        case Map:
+          unwrappedList.add(i, readableArray.getMap(i).toJavaMap());
+          break;
+        case Array:
+          unwrappedList.add(i, ReadableNativeArray.toJavaArray(readableArray.getArray(i)));
+          break;
+        default:
+          throw new IllegalArgumentsException("Could not convert object at index " + i + ".");
```

Comment:
@satya164 mind pointing to an example of this, or referencing the exception type that I can further look into?